BigFileMergeSort/Arrays.java

```diff
@@ -0,0 +1,192 @@
+package ua.goit.alg;
+
+import java.io.DataOutputStream;
+import java.io.EOFException;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.util.ArrayList;
+import java.util.NoSuchElementException;
+import java.util.Scanner;
+
+public class Arrays {
+
+    public static File mergeSort(File file, int blocksize) throws IOException {
+	ArrayList<File> partList = splitSort(file, blocksize);
+	File result = new File("result.txt");
+	result = mergeParts(partList).get(0);
+	return result;
+    }
+
+    public static ArrayList<File> mergeParts(ArrayList<File> partList) throws IOException {
+	if (partList.size() ==1) {
+	    return partList;
+	}
+	if (partList.size() > 1) {
+	    ArrayList<File> resultList = new ArrayList<File>();
+	    if (partList.size() % 2 == 1) {
+		File merged = mergeFiles(partList.get(0), partList.get(1));
+		partList.add(merged);
+		partList.remove(1);
+		partList.remove(0);
+	    }
+	    for (int i = 0; i < partList.size()-1; i++) {
+		resultList.add(mergeFiles(partList.get(i), partList.get(i+1)));
+		i++;
```

Comment:
(!) that's a no-no! modifying cycle counter inside of the cycle is unexpected thing in most cases.
instead do `for (int i = 0; i < partList.size() - 1; i+=2) {`